src/coreclr/jit/emitxarch.cpp

```diff
@@ -2651,22 +2658,64 @@ emitter::instrDesc* emitter::emitNewInstrAmdCns(emitAttr size, ssize_t dsp, int
     }
 }
 
-/*****************************************************************************
- *
- *  The next instruction will be a loop head entry point
- *  So insert a dummy instruction here to ensure that
- *  the x86 I-cache alignment rule is followed.
- */
-
-void emitter::emitLoopAlign()
+//-----------------------------------------------------------------------------
+//
+//  The next instruction will be a loop head entry point
+//  So insert a dummy instruction here to ensure that
+//  the x86 I-cache alignment rule is followed.
+//
+void emitter::emitLoopAlign(unsigned short paddingBytes)
 {
     /* Insert a pseudo-instruction to ensure that we align
        the next instruction properly */
 
-    instrDesc* id = emitNewInstrSmall(EA_1BYTE);
-    id->idIns(INS_align);
-    id->idCodeSize(15); // We may need to skip up to 15 bytes of code
-    emitCurIGsize += 15;
+    paddingBytes       = min(paddingBytes, 15); // We may need to skip up to 15 bytes of code
+    instrDescAlign* id = emitNewInstrAlign();
+    id->idCodeSize(paddingBytes);
+    emitCurIGsize += paddingBytes;
+
+    id->idaIG = emitCurIG;
+
+    /* Append this instruction to this IG's jump list */
+    id->idaNext        = emitCurIGAlignList;
+    emitCurIGAlignList = id;
+
+    /* Record the last IG that has align instruction */
+    emitLastAlignedIgNum = emitCurIG->igNum;
+}
+
+//-----------------------------------------------------------------------------
+//
+//  The next instruction will be a loop head entry point
+//  So insert a dummy instruction here to ensure that
+//  the x86 I-cache alignment rule is followed.
+//
+//  This emits more than one `INS_align` instruction depending on the
+//  alignmentBoundary parameter.
+//
+void emitter::emitLongLoopAlign(unsigned short alignmentBoundary)
+{
+    unsigned short nPaddingBytes    = alignmentBoundary - 1;
+    unsigned short nAlignInstr      = (nPaddingBytes + (15 - 1)) / 15;
+    unsigned short instrDescSize    = nAlignInstr * sizeof(instrDescAlign);
+    unsigned short insAlignCount    = nPaddingBytes / 15;
+    unsigned short lastInsAlignSize = nPaddingBytes % 15;
```

Comment:
Ditto here -- is there a symbolic constant for the max instr desc encoded size? If so we should use it instead of `15`.